src/client/dfs/dfs.c

```diff
@@ -2138,6 +2138,12 @@ dfs_remove(dfs_t *dfs, dfs_obj_t *parent, const char *name, bool force,
 	if (!exists)
 		D_GOTO(out, rc = ENOENT);
 
+	/* Check if the file is the expected one */
```

Comment:
this should actually return the oid of the removed object, not the user pass it in..
so you cannot do that here.